vignettes/compareToBaseR.Rmd

```diff
@@ -0,0 +1,683 @@
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>"
+)
+```
+
+This vignette aims to compare stringr functions to base R equivalents to facilitate users transitioning from using base R to stringr. Following the groupings of functions from the [stringr cheat sheet](https://github.com/rstudio/cheatsheets/blob/master/strings.pdf), these are the base equivalents to stringr functions.
+
+# Detect matches
+
+| base | stringr    |
+|------|------------|
+|   `grepl(pattern, string)`   | `str_detect(string, pattern)` |
+|   `grep(pattern, string)`   | `str_which(string, pattern)`  |
+|   `gregexpr(pattern, string)`   | `str_count(string, pattern)`  |
+|   `regexpr(pattern, string)`   | `str_locate(string, pattern)`  |
+|   `gregexpr(pattern, string)`    | `str_locate_all(string, pattern)` |
+
+# Subset strings
+
+| base | stringr    |
+|------|------------|
+|   `substr(string, start, end)`   | `str_sub(string, start, end)`    |
+|    `grep(pattern, string, value = TRUE)`  | `str_subset(string, pattern)` |
+|   `regexpr(pattern, string)` + `regmatches()`   | `str_extract(string, pattern)`|
+|   `regexec(patter, string)` + `regmatches()`   | `str_match(string, pattern)`  |
+
+
+# Manage lengths
+
+| base | stringr    |
+|------|------------|
+|   `nchar()`   | `str_length()` |
+|  N/A    | `str_pad()`    |
+|   N/A  | `str_trunc()`  |
+|   `trimws()`   | `str_trim()`   |
+|  `strwrap()`    | `str_wrap()`     |
+
+# Mutate strings
+
+| base | stringr        |
+|------|----------------|
+|  `substr(string, start, end)`    | `str_sub(string, start, end)`        |
+|   `sub(pattern, replacement, string)`   | `str_replace(string, pattern, replacement)`    |
+|   `gsub(pattern, replacement, string)`   | `str_replace_all(string, pattern, replacement)`|
+|   `tolower()`   | `str_to_lower()`   |
+|   `toupper()`   | `str_to_upper()`   |
+|  `tools::toTitleCase()`    | `str_to_title()`   |
+
+
+# Join and split
+
+| base | stringr        |
+|------|----------------|
+|   `paste()`   | `str_c()`          |
+|  `replicate()` + `paste()`    | `str_dup(string, times)`        |
+|  `strsplit(string, pattern)`    | `str_split_fixed(string, pattern)`|
+|   `paste()` + `eval()`   | `str_glue()`       |
+|    `paste()`  | `str_glue_data()`  |
+
+# Order strings
+
+| base | stringr        |
+|------|----------------|
+|   `order()`   | `str_order()`      |
+|   `sort()`   | `str_sort()`       |
+
+# Helpers
+
+| base | stringr      |
+|------|--------------|
+|  `iconv()`    | `str_conv()`     |
+|  N/A    | `str_view()`     |
+|   N/A   | `str_view_all()` |
+
+
+```{r}
+library(stringr)
+```
+
+Now we will walk through each of the functions in the above tables, noting the similarities and differences between base R and stringr.
+
+These stringr examples come from the [stringr documentation](https://cran.r-project.org/web/packages/stringr/stringr.pdf) and here, they are contrasted with the analogous base R operation(s).
+
+
+# Detect matches
+
+## `str_detect`
```

Comment:
Maybe:

```
## `str_detect()`: Detect the presence or absence of a pattern in a string
```

(note the parentheses after the function name; that's house style)